tests/Microsoft.Maui.IntegrationTests/AndroidTemplateTests.cs

```diff
@@ -0,0 +1,90 @@
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Maui.IntegrationTests.Android;
+
+namespace Microsoft.Maui.IntegrationTests
+{
+	public class AndroidTemplateTests : BaseBuildTest
+	{
+		Emulator TestAvd = new Emulator();
+		string testPackage = "";
+
+		[OneTimeSetUp]
+		public void AndroidTemplateFxtSetUp()
+		{
+			if (TestEnvironment.IsMacOS && RuntimeInformation.OSArchitecture == Architecture.Arm64)
+				TestAvd.Abi = "arm64-v8a";
+
+			TestAvd.InstallAvd();
+		}
+
+		[SetUp]
+		public void AndroidTemplateSetUp()
+		{
+			var emulatorLog = Path.Combine(TestDirectory, $"emulator-launch-{DateTime.UtcNow.ToFileTimeUtc()}.log");
+			Assert.IsTrue(TestAvd.LaunchAndWaitForAvd(720, emulatorLog), "Failed to launch Test AVD.");
+		}
+
+		[OneTimeTearDown]
+		public void AndroidTemplateFxtTearDown()
+		{
+			Adb.KillEmulator(TestAvd.Id);
+
+			// adb.exe can lock certain files on windows, kill it after tests complete
+			if (TestEnvironment.IsWindows)
+			{
+				Adb.Run("kill-server", deviceId: TestAvd.Id);
+				foreach (var p in Process.GetProcessesByName("adb.exe"))
+					p.Kill();
+			}
+		}
+
+		[TearDown]
+		public void AndroidTemplateTearDown()
+		{
+			Adb.UninstallPackage(testPackage);
+		}
+
+
+		[Test]
+		[TestCase("maui", "net6.0", "Debug")]
+		[TestCase("maui", "net6.0", "Release")]
+		[TestCase("maui", "net7.0", "Debug")]
+		[TestCase("maui", "net7.0", "Release")]
+		[TestCase("maui-blazor", "net6.0", "Debug")]
+		[TestCase("maui-blazor", "net6.0", "Release")]
+		[TestCase("maui-blazor", "net7.0", "Debug")]
+		[TestCase("maui-blazor", "net7.0", "Release")]
```

Comment:
Will we need a separate PR when this makes it to the `net8.0` branch?

Reply:
Yeah, we'll need an extra commit to replace `net6.0` with `net8.0` project types. We could try to generate this based on the content of some of the targets files but that could be done separately if it's even worthwhile.